gossip/src/cluster_info.rs

```diff
@@ -2778,49 +2778,80 @@ impl Node {
     ) -> Self {
         let (gossip_port, (gossip, ip_echo)) =
             Self::get_gossip_port(gossip_addr, port_range, bind_ip_addr);
-        let (tvu_port, tvu) = Self::bind(bind_ip_addr, port_range);
-        let (tvu_quic_port, tvu_quic) = Self::bind(bind_ip_addr, port_range);
-        let udp_config = SocketConfig { reuseport: false };
-        let quic_config = SocketConfig { reuseport: true };
+
+        let read_write_socket_config = SocketConfig::default();
+        let (tvu_port, tvu) =
+            Self::bind_with_config(bind_ip_addr, port_range, read_write_socket_config.clone());
+        let (tvu_quic_port, tvu_quic) =
+            Self::bind_with_config(bind_ip_addr, port_range, read_write_socket_config.clone());
+        let tpu_udp_config = SocketConfig::default().usage(SocketUsage::ReadOnly);
```

Comment:
in this PR i decided to set sockets read/write only (if they are). but setting this doesn't actually change the size of the socket buffer in this PR. Changing the buffer size will be done in a follow up PR

The other option could be to not set sockets as read/write only through the socket config builder. and wait to do that in the next PR when we actually change buffer sizes. Curious to hear thoughts on this.

Reply:
> The other option could be to not set sockets as read/write only through the socket config builder. and wait to do that in the next PR when we actually change buffer sizes. Curious to hear thoughts on this.

I probably still lean in favor of this approach. Limiting sockets to be `RO` or `WO` makes sense with the buffer capacity changes - "the read side is now marked as unused so let's also shrink the buffer to some small amount". Not sure how much work it'd be for you to split this up again

Reply:
shouldn't be too bad. stay tuned for an update lol. thank you for the help on this

Reply:
updated